### PR TITLE
Add monitoring server as minion

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

- remove Xen host
- add Monitoring server

For now, the monitoring server is using sumaform's "minion" module.
In a second phase, we might use a module of its own, possibly reusing existing "grafana" module. We'll see...

## Links:

Issue cards: https://github.com/SUSE/spacewalk/issues/19783 and https://github.com/SUSE/spacewalk/issues/17627
infrastructure: https://gitlab.suse.de/galaxy/infrastructure/-/merge_requests/663
main.tfs: https://github.com/SUSE/susemanager-ci/pull/716

### CODE REVIEW

File: modules/controller/variables.tf [hunk 1]
```diff
@@ -97,7 +97,7 @@ variable "kvmhost_configuration" {
   }
 }
 
-variable "xenhost_configuration" {
+variable "monitoringserver_configuration" {
   description = "use module.<VIRTHOST_NAME>.configuration, see main.tf.libvirt-testsuite.example"
```
Comment (cbosdo) [inline modules/controller/variables.tf:101]: Description probably needs a refresh too

File: salt/virthost/init.sls [hunk 2]
```diff
@@ -93,41 +81,6 @@ ifcfg-br0:
...
-    - source_hash: {{ grains['hvm_disk_image_hash'] }}
-    {% else %}
-    - skip_verify: True
-    {% endif %}
-    - mode: 655
-    - makedirs: True
-{% endif %}
-
-set_xen_default:
-  bootloader.grub_set_default:
```
Comment (cbosdo) [inline salt/virthost/init.sls:112]: You surely can also remove the python code of that state

**Review by cbosdo — CHANGES_REQUESTED** [review]
Some bits I am unsure about and more to delete

**Review by vandabarata — APPROVED** [review]

File: README_ADVANCED.md [hunk 3]
```diff
@@ -238,11 +237,6 @@ In order to use another or a cached image, use the `hvm_disk_image` variable.
 If the `hvm_disk_image` is set to the empty string, no image will be copied in `/var/testsuite-data/`.
 For example, to use a local image copy it in `salt/virthost/` folder and set `hvm_disk_image = "salt://virthost/imagename.qcow2"`
 
-The template disk image for Xen paravirtualized virtual machines is the openSUSE 15.1 JeOS image and is downloaded when applying the highstate on the Xen virtual host.
-As for the other image template, in order to use another or a cached image, use the `xen_disk_image` variable.
-
-Note that the Xen virtualization host will be rebooted when applying the highstate in order to use the newly installed Xen kernel.
```
Comment (srbarrios) [inline README_ADVANCED.md:244]: Same as above.

File: README_TESTING.md [hunk 4]
```diff
@@ -148,7 +148,7 @@ host_settings = {
   }
   kvm-host = {
   }
-  xen-host = {
```
Comment (srbarrios) [inline README_TESTING.md:151]: I would not replace that part, but just add a new one for the monitoring-server

File: modules/controller/main.tf [hunk 5]
```diff
@@ -37,17 +37,17 @@ module "controller" {
...
-    proxy         = var.proxy_configuration["hostname"]
-    client        = length(var.client_configuration["hostnames"]) > 0 ? var.client_configuration["hostnames"][0] : null
-    minion        = length(var.minion_configuration["hostnames"]) > 0 ? var.minion_configuration["hostnames"][0] : null
-    build_host    = length(var.buildhost_configuration["hostnames"]) > 0 ? var.buildhost_configuration["hostnames"][0] : null
-    redhat_minion = length(var.redhat_configuration["hostnames"]) > 0 ? var.redhat_configuration["hostnames"][0] : null
-    debian_minion = length(var.debian_configuration["hostnames"]) > 0 ? var.debian_configuration["hostnames"][0] : null
-    ssh_minion    = length(var.sshminion_configuration["hostnames"]) > 0 ? var.sshminion_configuration["hostnames"][0] : null
-    pxeboot_mac   = var.pxeboot_configuration["macaddr"]
-    kvm_host      = length(var.kvmhost_configuration["hostnames"]) > 0 ? var.kvmhost_configuration["hostnames"][0] : null
-    xen_host      = length(var.xenhost_configuration["hostnames"]) > 0 ? var.xenhost_configuration["hostnames"][0] : null
```
Comment (srbarrios) [inline modules/controller/main.tf:50]: That line would be okish to remove, as our controller should not support anymore xen, as our test framework can't handle it anymore. But all related to a deployment of a Xen module, should be kept, as still a supported module in Sumaform.

File: modules/cucumber_testsuite/main.tf [hunk 6]
```diff
@@ -283,48 +283,47 @@ module "kvm-host" {
   provider_settings = lookup(local.provider_settings_by_host, "kvm-host", {})
 }
 
-module "xen-host" {
-  source = "../virthost"
+module "monitoring-server" {
```
Comment (srbarrios) [inline modules/cucumber_testsuite/main.tf:286]: Please, keep in mind that we don't use cucumber_testsuite wrapper in our BV main.tf, and so, we need to support it also outside the cucumber_testsuite wrapper.

Comment (Bischoff) [reply modules/cucumber_testsuite/main.tf:286]: Correct. But this allows someone to use it if they want to experiment.

As a big fan of YAGNI, I will remove it if you insist.

Comment (Bischoff) [reply modules/cucumber_testsuite/main.tf:286]: After discussion, let's keep it as a convenience.

**Review by cbosdo — APPROVED** [review]

**Pull request merged**